regexp.go

```diff
@@ -33,14 +33,44 @@ func (e EmitterFunc) Emit(groups []string, lexer Lexer) Iterator { return e(grou
 func ByGroups(emitters ...Emitter) Emitter {
 	return EmitterFunc(func(groups []string, lexer Lexer) Iterator {
 		iterators := make([]Iterator, 0, len(groups)-1)
-		// NOTE: If this panics, there is a mismatch with groups. Uncomment the following line to debug.
+		// NOTE: If this panics, there is a mismatch with groups
 		for i, group := range groups[1:] {
 			iterators = append(iterators, emitters[i].Emit([]string{group}, lexer))
 		}
 		return Concaterator(iterators...)
 	})
 }
 
+// BySubBlockGroups uses the specified code and sublexerNameGroup from the
+// passed groups to emit tokens.
+func BySubBlockGroups(sublexerGetFunc func(string) Lexer, sublexerNameGroup, codeGroup int, emitters ...Emitter) Emitter {
```

Comment:
Rename this to `UsingByGroup()` as that more accurately reflects its intent. Also, can you describe in more detail what the parameters do?

Reply:
Ok.